types/d3/v3/index.d.ts

```diff
@@ -429,7 +433,7 @@ declare namespace d3 {
     /**
      * Administrivia: JavaScript primitive types, or "things that toString() predictably".
      */
-    export type Primitive = number | string | boolean;
+    export type Primitive = number | string | boolean | Date | undefined;
```

Comment:
I worry about this change, for two reasons:

1. If you're making changes for strictNullChecks purposes, then you should also add `null` to this list.
2. Why allow Date objects here? The reason we have number | string | boolean is to allow simple and relatively common cases of attributes like SVG's `stroke-width`. I'm not aware of any HTML or SVG attributes that would benefit from having an automatically-stringified Date object. If it's intentional, you can do the string conversion explicitly.

Reply:
I understand your points and can remove the addition. Nevertheless it is in line with the comment above the code saying: "things that toString() predictably" which is the case for Date and undefined. It happens in my project that I do things like: `.attr("title", (d) => d.date)` where  Typescript complains about the Primitive. When the code runs without problems.

Reply:
Ah, the title attribute isn't one I had thought about. Still, I'd prefer that you do `.attr('title', d => String(d.date))` because it's more obvious -- there are a bunch of other cases that I hesitate to allow because it makes it easier to make mistakes: consider using a `number[][]` for an SVG polyline/polygon, for example.

This kind of work is a delicate balance, and I err on the side of disallowing implicit conversions. You couldn't call `HTMLElement#setAttribute` with Date objects since the DOM API only really traffics in strings, and that means that TypeScript disallows it. I realize that it does cause some consternation, but this is the price of strongly typing an extremely flexible language.